node/pkg/driver/node_utils.go

```diff
@@ -347,3 +350,34 @@ func (n NodeUtils) IsNotMountPoint(file string) (bool, error) {
 func (n NodeUtils) GetPodPath(origPath string) string {
 	return path.Join(PrefixChrootOfHostRoot, origPath)
 }
+
+func (n NodeUtils) GenerateNodeID(hostName string, fcWWNs []string, iscsiIQN string) (string, error) {
+	var nodeId strings.Builder
+	nodeId.Grow(MaxNodeIdLength)
+	nodeId.WriteString(hostName)
+	nodeId.WriteString(NodeIdDelimiter)
+
+	if len(fcWWNs) != 0 {
+		if nodeId.Len()+len(fcWWNs[0]) <= MaxNodeIdLength {
+			nodeId.WriteString(fcWWNs[0])
+		} else {
+			return "", fmt.Errorf(ErrorWhileTryingToGenerateNodeId, nodeId.String(), MaxNodeIdLength)
+		}
+
+		for _, fcPort := range fcWWNs[1:] {
+			if nodeId.Len()+len(NodeIdFcDelimiter)+len(fcPort) <= MaxNodeIdLength {
+				nodeId.WriteString(NodeIdFcDelimiter)
+				nodeId.WriteString(fcPort)
+			}
+		}
+	}
+
+	if nodeId.Len()+len(NodeIdDelimiter)+len(iscsiIQN) <= MaxNodeIdLength {
+		nodeId.WriteString(NodeIdDelimiter)
+		nodeId.WriteString(iscsiIQN)
+	} else if len(fcWWNs) == 0 {
+		return "", fmt.Errorf(ErrorWhileTryingToGenerateNodeId, nodeId.String(), MaxNodeIdLength)
```

Comment:
if no ports are **found**, then I don't think a message of "could not **fit** any ports" is relevant

Reply:
1. yes
2. we have no UT at all right now.

Reply:
then either add unit tests, write a ticket number to add them, or get a 3rd opinion regarding the need for them

Reply:
I will add them, just after approval of the chanages..

Reply:
done